packages/nextjs/src/config/loaders/ast.ts

```diff
@@ -0,0 +1,219 @@
+  const nodesWithComments = ast.find(Node).filter(path => !!path.node.comments);
+  nodesWithComments.forEach(path => (path.node.comments = null));
+}
+
+/**
+ * Find an unused identifier name in the AST by repeatedly adding underscores to the beginning of the given original
+ * name until we find one which hasn't already been taken.
+ *
+ * @param userAST The AST to search
+ * @param origName The original name we want to alias
+ * @returns
+ */
+export function findAvailibleAlias(userAST: AST, origName: string): string {
+  let foundAvailableName = false;
+  let newName = origName;
+
+  while (!foundAvailableName) {
+    // Prefix the original function name (or the last name we tried) with an underscore and search for identifiers with
+    // the new name in the AST
+    newName = `_${newName}`;
+    const existingIdentifiers = findIdentifiers(userAST, newName);
+
+    // If we haven't found anything, we're good to go
+    foundAvailableName = existingIdentifiers.length === 0;
+  }
+
+  return newName;
+}
+
+/**
+ * More precise version of `identifierFilter`, used in `findIdentifiers`, which accounts for context. See note in
+ * `findIdentifiers` above.
+ */
+
+// const {
+//   AssignmentExpression,
+//   CallExpression,
+//   ExportSpecifier,
+//   FunctionDeclaration,
+//   Identifier,
+//   MemberExpression,
+//   Node,
+//   Property,
+//   ReturnStatement,
+//   VariableDeclaration,
+//   VariableDeclarator,
+// } = jscs;
+//
+// const identifierFilter = function (path: ASTPath<Identifier>): boolean {
+//   const node = path.node;
+//   const parentPath = path.parent as ASTPath;
+//   const parent = parentPath.node;
+//
+//   const hasCorrectName = node.name === name;
+//
+//   // Check that the identifier is being used in a valid context, one in which we do in fact want to replace it.
+//   //
+//   // Note: There are a million ways identifiers can be used - this is just a subset, but it should hit 99% of cases.
+//   // If anyone every files an issue because we're doing an incomplete job of transforming their code, get a
+//   // representative sample from them and throw it into https://astexplorer.net/ or
+//   // https://rajasegar.github.io/ast-finder/ (making sure in either case to set the parser to `recast`) to figure out
+//   // what to add below. (Find the `Identifier` node and note its parent's `type` value and the name of the key under
+//   // which it lives.) Note that neither tool seems to be able to handle the `export` keyword for some reason, but for
+//   // anything other than the case already included below, `ExportSpecifier` will be at least the grandparent; given
+//   // that we only care about recognizing the parent, we can just remove `export` from the sample code and it won't
+//   // make any difference to the part we care about.
+//   //
+//   // In all of the examples in the comments below, the identifer we're interested in is `someFunc`.
+//   const contextIsValid =
```

Comment:
Forgot to mention, do you think we should handle the case where users do `import { getServerSideProps } from '...'`? The way we replace identifiers right now would break imports like these.

Reply:
Yup, on it.